mitodl/open-discussions
# Display the most relevant course run, with dropdown, in course drawer

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #2124 

#### What's this PR do?
- Sends the most relevant course run id (if any, based on availability facets) to the drawer, where that run's info will be displayed (start date, price, instructors).  
- Adds a dropdown to the drawer allowing the user to select another course run (which will again update the drawer info shown).
- Does not restyle the dropdown to match the design, I'm assuming that restyling is a separate PR.

#### How should this be manually tested?
Do some course searches, with variations on the availability facets used.  Click on one of the search results to open the drawer.  The run shown in the drawer by default should be the best match for the availabilities selected.  There should be a dropdown at the top where you can choose different runs, and doing so should update the start date, instructors, and prices shown in the drawer.

OCW courses should have semester/year in the dropdown, MITx courses should have dates in the dropdown in a format like "September 10, 2019".
Bootcamps and any course with only 1 run should have a plain div with the date instead of a dropdown.


#### Screenshots (if appropriate)

OCW:
<img width="496" alt="Screen Shot 2019-09-11 at 8 55 09 AM" src="https://user-images.githubusercontent.com/187676/64701782-6b6ccd00-d477-11e9-99d8-6da51f8ff446.png">

OCW (1 run):
<img width="496" alt="Screen Shot 2019-09-11 at 9 36 33 AM" src="https://user-images.githubusercontent.com/187676/64701989-c1417500-d477-11e9-9070-a4c4b5038806.png">



MITx:
<img width="496" alt="Screen Shot 2019-09-11 at 9 33 25 AM" src="https://user-images.githubusercontent.com/187676/64702025-d0282780-d477-11e9-89c3-1fc79db67d97.png">



Bootcamp (1 run):
<img width="498" alt="Screen Shot 2019-09-11 at 8 56 41 AM" src="https://user-images.githubusercontent.com/187676/64702229-26956600-d478-11e9-8324-1986fa2f9265.png">
